stix_shifter/stix_transmission/src/modules/elastic_ecs/api_client.py

```diff
@@ -0,0 +1,101 @@
+import base64
+from ..utils.RestApiClient import RestApiClient
+import json
+import re
+
+class APIClient():
+
+    PING_ENDPOINT = '_cluster/health?pretty'
+
+    def __init__(self, connection, configuration):
+        headers = dict()
+        auth = configuration.get('auth')
+        self.indices = configuration.get('elastic_ecs', {}).get('indices', None)
+        if isinstance(self.indices, list):  # Get list of all indices
+            self.indices = ",".join(self.indices)
+        if self.indices:
+            self.endpoint = self.indices + '/' +'_search'
+        else:
+            self.endpoint = '_search'
+        if auth:
+            headers['Authorization'] = b"Basic " + base64.b64encode(
```

Comment:
To avoid repeating code, you may wish to set up the required parameters in the `if auth: else:` block and do just the one RestApiClient call. You can set the `url_modifier_function` to none if it's not needed.

Reply:
Yes, I will update the code.

Reply:
Updated code to remove repeating code. Ref: c6aee8a
Please review and let me know your comments.